argsort regression

test code (a.py):
```py
import numpy as np

#pythran export foo(float[:,:])
def foo(data: np.ndarray) -> np.ndarray:
    return data.argsort()
```

with 0.14.0:
```shell
(venv) spacetanuki% pythran a.py
(venv) spacetanuki% 
```

with 0.18.0:
```shell
(venv) spacetanuki% pythran a.py
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:16:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/argsort.hpp:79:30: error: reference to 'ndarray' is ambiguous
   79 |     return _argsort(a, axis, ndarray::quicksorter());
      |                              ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:16:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/argsort.hpp:87:32: error: reference to 'ndarray' is ambiguous
   87 |       return _argsort(a, axis, ndarray::mergesorter());
      |                                ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:16:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/argsort.hpp:89:32: error: reference to 'ndarray' is ambiguous
   89 |       return _argsort(a, axis, ndarray::heapsorter());
      |                                ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:16:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/argsort.hpp:91:32: error: reference to 'ndarray' is ambiguous
   91 |       return _argsort(a, axis, ndarray::stablesorter());
      |                                ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:16:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/argsort.hpp:92:30: error: reference to 'ndarray' is ambiguous
   92 |     return _argsort(a, axis, ndarray::quicksorter());
      |                              ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:17:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray.hpp:22:61: error: redefinition of 'ndarray' as different kind of symbol
   22 |   types::ndarray<typename dtype::type, types::pshape<long>> ndarray(long size,
      |                                                             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: previous definition is here
   19 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:17:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray.hpp:25:12: error: reference to 'ndarray' is ambiguous
   25 |     return ndarray(types::pshape<long>(size), d);
      |            ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray.hpp:16:3: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   16 |   ndarray(pS const &shape, dtype)
      |   ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray.hpp:19:3: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   ndarray(long size, dtype d = dtype());
      |   ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray.hpp:24:3: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   24 |   ndarray(std::integral_constant<long, N>, dtype d = dtype());
      |   ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:17:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray.hpp:31:3: error: redefinition of 'ndarray' as different kind of symbol
   31 |   ndarray(std::integral_constant<long, N>, dtype d)
      |   ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: previous definition is here
   10 |   namespace ndarray
      |             ^
In file included from /var/folders/74/hw1sphgx0lv63q6pq_n5grw00000gn/T/tmp3k3936_f.cpp:17:
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray.hpp:33:12: error: reference to 'ndarray' is ambiguous
   33 |     return ndarray(types::pshape<std::integral_constant<long, N>>({}), d);
      |            ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray.hpp:24:3: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   24 |   ndarray(std::integral_constant<long, N>, dtype d = dtype());
      |   ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray/sort.hpp:10:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   10 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray.hpp:16:3: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   16 |   ndarray(pS const &shape, dtype)
      |   ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/numpy/ndarray/sort.hpp:19:13: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   namespace ndarray
      |             ^
/Users/yamamoto/git/garbage/py/np2/venv/lib/python3.12/site-packages/pythran/pythonic/include/numpy/ndarray.hpp:19:3: note: candidate found by name lookup is '(anonymous namespace)::pythonic::numpy::ndarray'
   19 |   ndarray(long size, dtype d = dtype());
      |   ^
9 errors generated.
WARNING: Compilation error, trying hard to find its origin...
Compilation error, trying hard to find its origin...
WARNING: Nope, I'm going to flood you with C++ errors!
Nope, I'm going to flood you with C++ errors!
CRITICAL: Cover me Jack. Jack? Jaaaaack!!!!
E: error: command '/usr/bin/clang++' failed with exit code 1
Cover me Jack. Jack? Jaaaaack!!!!
E: error: command '/usr/bin/clang++' failed with exit code 1
(venv) spacetanuki% 
```

environment:
```
spacetanuki% sw_vers
ProductName:            macOS
ProductVersion:         15.6
BuildVersion:           24G84
spacetanuki% uname -a
Darwin spacetanuki.lan 24.6.0 Darwin Kernel Version 24.6.0: Mon Jul 14 11:28:17 PDT 2025; root:xnu-11417.140.69~1/RELEASE_X86_64 x86_64
spacetanuki% xcodebuild -version
Xcode 16.4
Build version 16F6
spacetanuki% clang++ --version
Apple clang version 17.0.0 (clang-1700.0.13.5)
Target: x86_64-apple-darwin24.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
spacetanuki% python --version
Python 3.13.5
spacetanuki% 
```

pythran installed with pip.

--- a/pythran/pythonic/include/numpy/ndarray.hpp
+++ b/pythran/pythonic/include/numpy/ndarray.hpp
@@ -10,20 +10,25 @@ PYTHONIC_NS_BEGIN
 namespace numpy
 {
 
-  template <class pS, class dtype = functor::float64>
-  types::ndarray<typename dtype::type, sutils::shape_t<pS>>
-  ndarray(pS const &shape, dtype d = dtype());
+  namespace anonymous
+  {
 
-  template <class dtype = functor::float64>
-  types::ndarray<typename dtype::type, types::pshape<long>>
-  ndarray(long size, dtype d = dtype());
+    template <class pS, class dtype = functor::float64>
+    types::ndarray<typename dtype::type, sutils::shape_t<pS>>
+    ndarray(pS const &shape, dtype d = dtype());
 
-  template <long N, class dtype = functor::float64>
-  types::ndarray<typename dtype::type,
-                 types::pshape<std::integral_constant<long, N>>>
-  ndarray(std::integral_constant<long, N>, dtype d = dtype());
+    template <class dtype = functor::float64>
+    types::ndarray<typename dtype::type, types::pshape<long>>
+    ndarray(long size, dtype d = dtype());
 
-  DEFINE_FUNCTOR(pythonic::numpy, ndarray);
+    template <long N, class dtype = functor::float64>
+    types::ndarray<typename dtype::type,
+                   types::pshape<std::integral_constant<long, N>>>
+    ndarray(std::integral_constant<long, N>, dtype d = dtype());
+
+  } // namespace anonymous
+
+  DEFINE_FUNCTOR(pythonic::numpy::anonymous, ndarray);
 } // namespace numpy
 PYTHONIC_NS_END
 

--- a/pythran/pythonic/numpy/ndarray.hpp
+++ b/pythran/pythonic/numpy/ndarray.hpp
@@ -11,27 +11,33 @@ PYTHONIC_NS_BEGIN
 
 namespace numpy
 {
-  template <class pS, class dtype>
-  types::ndarray<typename dtype::type, sutils::shape_t<pS>>
-  ndarray(pS const &shape, dtype)
-  {
-    return {(sutils::shape_t<pS>)shape, builtins::None};
-  }
 
-  template <class dtype>
-  types::ndarray<typename dtype::type, types::pshape<long>> ndarray(long size,
-                                                                    dtype d)
+  namespace anonymous
   {
-    return ndarray(types::pshape<long>(size), d);
-  }
 
-  template <long N, class dtype>
-  types::ndarray<typename dtype::type,
-                 types::pshape<std::integral_constant<long, N>>>
-  ndarray(std::integral_constant<long, N>, dtype d)
-  {
-    return ndarray(types::pshape<std::integral_constant<long, N>>({}), d);
-  }
+    template <class pS, class dtype>
+    types::ndarray<typename dtype::type, sutils::shape_t<pS>>
+    ndarray(pS const &shape, dtype)
+    {
+      return {(sutils::shape_t<pS>)shape, builtins::None};
+    }
+
+    template <class dtype>
+    types::ndarray<typename dtype::type, types::pshape<long>> ndarray(long size,
+                                                                      dtype d)
+    {
+      return ndarray(types::pshape<long>(size), d);
+    }
+
+    template <long N, class dtype>
+    types::ndarray<typename dtype::type,
+                   types::pshape<std::integral_constant<long, N>>>
+    ndarray(std::integral_constant<long, N>, dtype d)
+    {
+      return ndarray(types::pshape<std::integral_constant<long, N>>({}), d);
+    }
+
+  } // namespace anonymous
 } // namespace numpy
 PYTHONIC_NS_END
 

--- a/pythran/tests/test_numpy_func2.py
+++ b/pythran/tests/test_numpy_func2.py
@@ -621,7 +621,7 @@ def test_copy0(x):
         self.run_test("def np_around5(x): from numpy import around ; return around(x,-1)", [1,2,3,11], np_around5=[List[int]])
 
     def test_argsort0(self):
-        self.run_test("def np_argsort0(x): from numpy import argsort ; return argsort(x)", numpy.array([3, 1, 2]), np_argsort0=[NDArray[int,:]])
+        self.run_test("import numpy\ndef np_argsort0(x: numpy.ndarray) -> numpy.ndarray: from numpy import argsort ; return argsort(x)", numpy.array([3, 1, 2]), np_argsort0=[NDArray[int,:]])
 
     def test_argsort1(self):
         self.run_test("def np_argsort1(x): return x.argsort()", numpy.array([[3, 1, 2], [1 , 2, 3]]), np_argsort1=[NDArray[int,:,:]])

